docs/examples/request_data/test_request_data_10.py

```diff
@@ -0,0 +1,63 @@
+import io
+from hashlib import sha256
+from typing import Any, Dict, List, Tuple
+
+from typing_extensions import Annotated
+
+from litestar import Litestar, post
+from litestar.datastructures import UploadFile
+from litestar.enums import RequestEncodingType
+from litestar.params import Body
+from litestar.testing import TestClient
+
+
+@post(path="/")
+async def handle_file_upload(
+    data: Annotated[List[UploadFile], Body(media_type=RequestEncodingType.MULTI_PART)],
+) -> Dict[str, Tuple[str, str, Any]]:
+    result = {}
+
+    for file in data:
+        content = await file.read()
+        result[file.filename] = (sha256(content).hexdigest(), file.content_type, file.headers)
+
+    return result
+
+
+app = Litestar(route_handlers=[handle_file_upload])
+
+
+def test_file_upload() -> None:
+    with TestClient(app) as client:
+        # if you pass a dict to the `files` parameter without specifying a filename, it will default to `upload`
+        #     # file (or bytes)
```

Comment:
httpx has 2x 4-ways of passing stuff to the `files` kwarg, dict-like-way or tuple-like-way  and for each 4 possibilities.
the test explores the 4 possibilities and adds some comments on "weirdness" that could happen, I put them here, but I'm not sure if it should be in docs, maybe it's too precise,
I wanted to add that because I've been bitten by it, the `upload` default filename is definitely not obvious.